server/sonar-db-migration/src/test/java/org/sonar/server/platform/db/migration/version/v64/CreateMembersGroupsInEachOrganizationTest.java

```diff
@@ -0,0 +1,317 @@
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+package org.sonar.server.platform.db.migration.version.v64;
+
+import java.sql.SQLException;
+import java.util.Date;
+import java.util.List;
+import javax.annotation.Nullable;
+import org.assertj.core.groups.Tuple;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.sonar.api.utils.System2;
+import org.sonar.api.utils.log.LogTester;
+import org.sonar.core.util.stream.Collectors;
+import org.sonar.db.CoreDbTester;
+
+import static java.lang.String.format;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.tuple;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public class CreateMembersGroupsInEachOrganizationTest {
+
+  private static final Date PAST = new Date(100_000_000_000L);
+  private static final Date NOW = new Date(500_000_000_000L);
+  private static final String DEFAULT_ORGANIZATION_UUID = "def-org";
+  private static final String ORGANIZATION_1 = "ORGANIZATION_1";
+  private static final String ORGANIZATION_2 = "ORGANIZATION_2";
+  private static final String TEMPLATE_1 = "TEMPLATE_1";
+  private static final String TEMPLATE_2 = "TEMPLATE_2";
+
+  @Rule
+  public ExpectedException expectedException = ExpectedException.none();
+
+  @Rule
+  public LogTester logTester = new LogTester();
+
+  @Rule
+  public CoreDbTester db = CoreDbTester.createForSchema(CreateMembersGroupsInEachOrganizationTest.class, "initial.sql");
+
+  private System2 system2 = mock(System2.class);
+
+  private CreateMembersGroupsInEachOrganization underTest = new CreateMembersGroupsInEachOrganization(db.database(), system2);
+
+  @Before
+  public void setUp() throws Exception {
+    when(system2.now()).thenReturn(NOW.getTime());
+  }
+
+  @Test
+  public void does_nothing_when_organization_disabled() throws Exception {
+    setupDefaultOrganization();
+    insertOrganization(ORGANIZATION_1, TEMPLATE_1);
+    insertOrganization(ORGANIZATION_2, TEMPLATE_2);
+    insertPermissionTemplate(ORGANIZATION_1, TEMPLATE_1, "Default");
+    insertPermissionTemplate(ORGANIZATION_2, TEMPLATE_2, "Default");
+
+    underTest.execute();
+
+    checkGroups();
```

Comment:
suggestion: create a specific method checkNoGroups and checkNoPermTemplateGroups